src/requests/BaseApi.ts

```diff
@@ -0,0 +1,168 @@
+            }
+        }
+        // if all query parameters are present in the query params variable then we can replace them with our parameter from the template
+        let currentPath = this.baseClass.basePath + this.currentRequest.path;
+
+        if (this.currentQueryParams.length > 0) {
+            // check for all required parameters from this.currentRequest.requestQueryParams
+            for (const param of this.currentQueryParams) {
+                currentPath = currentPath.replace(`{${param.name}}`, param.value);
+            }
+            // console.log(currentPath);
+            
+            if (currentPath.includes("{") || currentPath.includes("}")) {
+                throw new Error("Missing required query parameter!");
+            }
+        }
+
+        // now we add any path parameters
+        if (pathKeys) {
+            currentPath += "?" + pathKeys;
+        }
+
+        const url: URL = new URL(currentPath);
+
+        // check headers
+        const headers: headerOptions = {
+            "User-Agent": this.baseClass.headerAgent // we set the User_Agent header
+        }
+        // headers["User-Agent"] = this.baseClass.headerAgent; 
+
+        if (!this.currentRequest.cookiesNeeded.includes("none")) {
+            headers["Cookie"] = ``;
+
+            if (this.currentRequest.cookiesNeeded.includes("authorization")) {
+                headers['Authorization'] = `Basic ${this.baseClass.getBase64Credentials()}`;
+            }
+            if (this.currentRequest.cookiesNeeded.includes("authCookie")) {
+                headers["Cookie"] += `${this.baseClass.instanceCookie.getAuthCookie()} `;
+            }
+            if (this.currentRequest.cookiesNeeded.includes("twoFactorAuth")) {
+                headers["Cookie"] += `${this.baseClass.instanceCookie.getTwoFactorAuthCookie()} `;
+            }
+        }
+        
+        // check options
+        const options: VRCRequest = {
+            method: this.currentRequest.method,
+            headers: headers
+        };
+        if (body && this.currentRequest.requiresData) {
+            options.body = JSON.stringify(body);
+        }
+
+        // call the fetch
+        // console.log("-- BEFORE CALL --");
+        // console.log("URL: ", url);
+        // console.log("options: ", options);
+        // console.log("-- END BEFORE CALL --");
+        
+
+        try {
+            const response: API<E | RequestSuccess, RequestError> = await fetch(url, { ...options });
+            
+            if (!response.ok) {
+                throw new Error(`HTTP error! Status: ${response.status}`);
+            }
+
+            const data = await response.json() as E;
+            
+            return data;
+
+        } catch (err) {
+            console.error("Error executing request: ", err);
+            if (err instanceof Error) {
+                throw new Error(`Error executing request: ${err.message}`);
+            }
+
+        }
+        return {} as E;
```

Comment:
It wouldn't be smart to return a empty body on the request, rather if anything goes wrong return undefined or just throw a error.